packages/modal/src/modal.ts

```diff
@@ -36,22 +35,51 @@ class Modal {
         const { showModal, hideModal, autoInitModal, isHidden } = initModal(modalElement, triggerButton, this.options);
 
         if (this.state === "open") {
-            showModal()
+            this.playOpenAnimation(); // Use animation when opening the modal
         }
-        autoInitModal()
-        this.showModal = showModal
-        this.hideModal = hideModal
-        this.isHidden = isHidden
+
+        this.showModal = this.playOpenAnimation.bind(this); // Update to use the open animation
+        this.hideModal = this.playCloseAnimation.bind(this); // Update to use the close animation
+        this.isHidden = isHidden;
+        
+        autoInitModal();
     }
 
+    // Predefined open animation
+    private playOpenAnimation(): void {
+        this.modalElement.style.display = "block"; // Ensure it's visible before animation
```

Comment:
We don’t handle this part; it's up to the user to decide how they want to manage the modal's appearance using data attributes. There's a `data-state` attribute that indicates whether the modal is visible or not, which users can control with CSS. 

Take a look at the [examples in the documentation](https://flexilla-docs.vercel.app/docs/components/modal#example) to see what I mean.